File: LIS.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& a) {
        int n=a.size();
        if(n==0)
            return 0;
        int dp[n+1];
        for(int i=0;i<=n;i++)
            dp[i]=1;
        for(int i=1;i<n;i++)
        {
            for(int j=0;j<i;j++)
            {
                if(a[i]>a[j])
                    dp[i]=max((dp[j]+1),dp[i]);
            }
        }
        
        int mx=0;
        for(int i=0;i<=n;i++)
            mx=max(dp[i],mx);
        return mx;
    }
};

File: LIS.py
def LongestIncreasingSubsequence(a):
  n=len(a)
  if n==0:
    return 0
  dp=[]
  
  for i in range(n+1):
    dp.append(1)
    
    
  for i in range(1,n):
    for j in range(i):
      if a[i]>a[j]:
        dp[i]=max(dp[i],(dp[j]+1))
        
  mx=0
  
  
  for z in range(n+1):
    mx=max(dp[i],mx)
  return mx;



if _name_ == "_main_":
  
  l=list(map(int, input("Enter an array :\n").split()))
  answer=LongestIncreasingSubsequence(l)
  print("Longest Increasing Subsequence :",answer"r)
